integration-test/http/http_system_test.c

```diff
@@ -0,0 +1,362 @@
+ * HTTP response headers and body.
+ */
+static uint8_t userBuffer[ USER_BUFFER_LENGTH ];
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Connect to HTTP server with reconnection retries.
+ *
+ * If connection fails, retry is attempted after a timeout.
+ * Timeout value will exponentially increase until maximum
+ * timeout value is reached or the number of attempts are exhausted.
+ *
+ * @param[out] pNetworkContext The output parameter to return the created network context.
+ *
+ */
+static void connectToServerWithBackoffRetries( NetworkContext_t * pNetworkContext );
+
+/**
+ * @brief Send an HTTP request based on a specified method and path, then
+ * print the response received from the server.
+ *
+ * @param[in] pTransportInterface The transport interface for making network calls.
+ * @param[in] pMethod The HTTP request method.
+ * @param[in] pPath The Request-URI to the objects of interest.
+ */
+static void sendHttpRequest( const TransportInterface_t * pTransportInterface,
+                             const char * pMethod,
+                             const char * pPath );
+
+/*-----------------------------------------------------------*/
+
+static void connectToServerWithBackoffRetries( NetworkContext_t * pNetworkContext )
+{
+    /* Status returned by the retry utilities. */
+    RetryUtilsStatus_t retryUtilsStatus = RetryUtilsSuccess;
+    /* Struct containing the next backoff time. */
+    RetryUtilsParams_t reconnectParams;
+    /* Status returned by OpenSSL transport implementation. */
+    OpensslStatus_t opensslStatus;
+
+    /* Reset or initialize file-scoped global variables. */
+    ( void ) memset( &opensslCredentials, 0, sizeof( opensslCredentials ) );
+    opensslCredentials.pRootCaPath = ROOT_CA_CERT_PATH;
+
+    serverInfo.pHostName = SERVER_HOST;
+    serverInfo.hostNameLength = SERVER_HOST_LENGTH;
+    serverInfo.port = HTTPS_PORT;
+
+    /* Initialize reconnect attempts and interval */
+    RetryUtils_ParamsReset( &reconnectParams );
+
+    /* Attempt to connect to HTTP server. If connection fails, retry after
+     * a timeout. Timeout value will exponentially increase until maximum
+     * attempts are reached.
+     */
+    do
+    {
+        /* Establish a TLS session, on top of TCP connection, with the HTTP server. */
+        opensslStatus = Openssl_Connect( pNetworkContext,
+                                         &serverInfo,
+                                         &opensslCredentials,
+                                         TRANSPORT_SEND_RECV_TIMEOUT_MS,
+                                         TRANSPORT_SEND_RECV_TIMEOUT_MS );
+
+        if( opensslStatus != OPENSSL_SUCCESS )
+        {
+            retryUtilsStatus = RetryUtils_BackoffAndSleep( &reconnectParams );
+        }
+
+        if( retryUtilsStatus == RetryUtilsRetriesExhausted )
+        {
+            LogDebug( ( "Connection to the server failed, all attempts exhausted." ) );
```

Comment:
LogError or Debug?

Reply:
Should be LogError, thanks for pointing that out.